src/components/templates/Camp/components/Map.tsx

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import { CampVenue } from "types/CampVenue";
+import { CampRoomData } from "types/CampRoomData";
+import { RoomAttendance } from "./RoomAttendance";
+import { isRoomValid } from "validation";
+
+import "./Map.scss";
+
+interface PropsType {
+  venue: CampVenue;
+  attendances: { [location: string]: number };
+  setSelectedRoom: (room: CampRoomData) => void;
+  setIsRoomModalOpen: (value: boolean) => void;
+}
+
+export const Map: React.FC<PropsType> = ({
+  venue,
+  attendances,
+  setSelectedRoom,
+  setIsRoomModalOpen,
+}) => {
+  if (!venue) {
+    return <>Loading map...</>;
+  }
+
+  const openRoomModal = (room: CampRoomData) => {
+    setSelectedRoom(room);
+    setIsRoomModalOpen(true);
+  };
+
+  return (
+    <>
+      <div id="map" className="map-container">
+        {venue.rooms.filter(isRoomValid).map((room, idx) => (
```

Comment:
if you notice slow performance, this could be put in a useMemo instead of recreating the array at every render